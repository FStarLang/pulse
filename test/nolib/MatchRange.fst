module MatchRange

#lang-pulse
open Pulse.Nolib

[@@expect_failure [19]]
fn completeness_err2 (x : list int)
{
  ();
  ();
  ();
  ();
  let hd::tl = x;
  ();
  ();
  ();
  ();
  ();
}

[@@expect_failure [19]]
fn completeness_err1 (x : list int)
{
  ();
  ();
  ();
  ();
  match x {
    hd :: tl -> {
      ();
    }
  };
}
