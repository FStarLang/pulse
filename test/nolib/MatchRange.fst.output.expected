>> Got issues: [
* Error 19 at MatchRange.fst(12,2-18,5):
  - Could not verify that this match is exhaustive.
  - Patterns are incomplete
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - Also see: MatchRange.fst(13,6-13,12)
  - Other related locations: dummy(0,0-0,0)

>>]
>> Got issues: [
* Error 19 at MatchRange.fst(28,2-32,3):
  - Could not verify that this match is exhaustive.
  - Patterns are incomplete
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - Also see: MatchRange.fst(28,8-28,9)
  - Other related locations: dummy(0,0-0,0)

>>]
